### PR TITLE
Bh/merge upstream

### PR DESCRIPTION
**Pull request by haerdib**
Fix merge changes according to the diff of this branch vs upstream master branch (https://github.com/integritee-network/worker/compare/master...litentry:merge-upstream?diff=unified)

### CODE REVIEW

File: app-libs/stf/Cargo.toml [hunk 1]
```diff
@@ -67,7 +67,7 @@ sp-runtime = { version = "4.0.0-dev", default-features = false, git = "https://g
 sc-keystore = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master", optional = true }
 
 # litentry
-sp-io = { default-features = false, features = ["disable_oom", "disable_panic_handler", "disable_allocator", "sgx"], git = "https://github.com/litentry/sgx-runtime", branch = "master", optional = true }
+sp-io = { default-features = false, features = ["disable_oom", "disable_panic_handler", "disable_allocator"], git = "https://github.com/litentry/sgx-runtime", branch = "master", optional = true }
```
Comment (haerdib) [inline app-libs/stf/Cargo.toml:70]: Fixes the sgx / std compile error

File: app-libs/stf/src/lib.rs [hunk 2]
```diff
@@ -48,10 +48,6 @@ pub type AuthorityId = <Signature as Verify>::Signer;
 pub type AccountId = AccountId32;
 pub type Hash = sp_core::H256;
 pub type BalanceTransferFn = ([u8; 2], AccountId, Compact<u128>);
-//FIXME: Is this really necessary to define all variables three times?
```
Comment (haerdib) [inline app-libs/stf/src/lib.rs:51]: Leftover of upstream merge

File: app-libs/stf/src/stf_sgx_primitives.rs [hunk 3]
```diff
@@ -17,11 +17,6 @@
 
 use codec::{Decode, Encode};
 use itp_types::H256;
-use sgx_tstd as std;
```
Comment (haerdib) [inline app-libs/stf/src/stf_sgx_primitives.rs:20]: Left over from upstream merge. Cause compile error.

File: enclave-runtime/Cargo.toml [hunk 4]
```diff
@@ -44,13 +44,13 @@ retain_mut = { version = "0.1.2"}
 derive_more = { version = "0.99.5" }
 byteorder = { version = "1.4.2", default-features = false }
 arrayvec = { version = "0.7.1", default-features = false }
-hex = { version = "0.4.3", default-features = false }
+hex = { version = "0.4.3", default-features = false, features = ["alloc"] }
```
Comment (haerdib) [inline enclave-runtime/Cargo.toml:47]: With this import the hex.rs file was able to deleted in upstream.

**Review by suinuj — APPROVED** [review]

**Pull request merged**